### PR TITLE
Fixing various contrib changes causing CASL breakage

### PR DESCRIPTION
**Pull request by oybed**
#### What does this PR do?
Fixing breakage found in CASL due to various changes in contrib. These are the changes:

heat_stack.yaml.j2:
- `k8s_type` set to new set of facts - changed to have default values for backwards compatibility  
- `image` set to new set of facts - changed to have a default value for backwards compatibility + provide more flexibility 

openstack-stack/defaults/main.yml:
- `provider_network` set to *False* instead of *None* to ensure conditionals are evaluated correctly 

The following PRs introduced these changes - please have the authors review the fixes in this PR:
https://github.com/openshift/openshift-ansible-contrib/pull/643
https://github.com/openshift/openshift-ansible-contrib/pull/637
https://github.com/openshift/openshift-ansible-contrib/pull/701

#### How should this be manually tested?
Ensure CASL provisioning works. 

#### Is there a relevant Issue open for this?
N/A

#### Who would you like to review this?
cc: @tomassedovic @bogdando @Tlacenka 

@etsauer FYI

### CODE REVIEW

File: roles/openstack-stack/defaults/main.yml [hunk 1]
```diff
@@ -18,4 +18,4 @@ dns_volume_size: 1
 lb_volume_size: 5
 use_bastion: False
 ui_ssh_tunnel: False
-provider_network: None
+provider_network: False
```
Comment (tomassedovic) [inline roles/openstack-stack/defaults/main.yml]: Why is this necessary? Jinja's conditionals treat `None` as falsey. What is this breaking for CASL?

Comment (oybed) [reply roles/openstack-stack/defaults/main.yml]: @tomassedovic I didn't capture the error, but running this with Ansible v2.3.1.x, the error was something like "provider_network 'None' not valid".

Comment (oybed) [reply roles/openstack-stack/defaults/main.yml]: i.e.: error coming from the Ansible modules, not CASL specific per-say

**Comment by oybed** [comment]
@tomassedovic yes, having a CI job for CASL would be great. We have discussed that for our own needs as well, so maybe we can leverage/re-use some of the CI configurations that you already have? If we can find a time to discuss, I'd love to get more details from you. Are you by a chance able to join our CASL call this Friday?

File: roles/openstack-stack/defaults/main.yml (hunk 1 above)
Comment (tomassedovic) [reply roles/openstack-stack/defaults/main.yml]: Hm, I'm running `2.3.1.0` and had never seen it. Might be a Jinja version or something. Whatever, I'll merge it.

**Comment by tomassedovic** [comment]
@oybed yeah that might work, though our current "CI" only has a capacity for one job, so we'll have to figure out where to run it. I'll try to make it to the call (much appreciate you moving it earlier in the day).
